src/lib.rs

```diff
@@ -1,9 +1,46 @@
 pub mod client_features;
 pub mod client_metrics;
 pub mod frontend;
+use std::{collections::HashSet, hash::Hash};
+
+pub trait Merge {
+    fn merge(self, other: Self) -> Self;
+}
+
+pub trait Deduplicate<T>
+where
+    T: Hash + Eq,
+{
+    fn deduplicate(self) -> Self;
+}
+
+impl<T> Deduplicate<T> for Vec<T>
+where
+    T: Hash + Eq,
+{
+    fn deduplicate(self) -> Self {
+        self.into_iter()
+            .collect::<HashSet<T>>()
+            .into_iter()
+            .collect()
+    }
+}
+
+impl<T> Merge for Vec<T>
+where
+    T: Hash + Eq,
```

Comment:
I would have assumed that `T: Deduplicate`, it's amazing that it can derive that because of being Hash+Eq it can also be deduplicated

Reply:
Yeah, neat hey?